test/widgets/dialog_test.dart

```diff
@@ -0,0 +1,98 @@
+import 'package:checks/checks.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/widgets/dialog.dart';
+
+import '../model/binding.dart';
+import 'dialog_checks.dart';
+import 'test_app.dart';
+
+void main() {
+  TestZulipBinding.ensureInitialized();
+
+  late BuildContext context;
+
+  const String title = "Dialog Title";
+  const String message = "Dialog message.";
+
+  Future<void> prepare(WidgetTester tester) async {
+    addTearDown(testBinding.reset);
+
+    await tester.pumpWidget(const TestZulipApp(
+      child: Scaffold(body: Placeholder())));
+    await tester.pump();
+    context = tester.element(find.byType(Placeholder));
+  }
+
+  group('showErrorDialog', () {
+    testWidgets('show error dialog', (tester) async {
+      await prepare(tester);
+
+      showErrorDialog(context: context, title: title, message: message);
+      await tester.pump();
+      checkErrorDialog(tester, expectedTitle: title, expectedMessage: message);
+
+    }, variant: const TargetPlatformVariant({TargetPlatform.android, TargetPlatform.iOS}));
+
+    testWidgets('user closes error dialog', (tester) async {
+      await prepare(tester);
+
+      showErrorDialog(context: context, title: title, message: message);
+      await tester.pump();
+
+      Widget button = checkErrorDialog(tester, expectedTitle: title);
+      await tester.tap(find.byWidget(button));
+      await tester.pump();
+      checkNoDialog(tester);
+    });
+  });
+
+  group('showSuggestedActionDialog', () {
+    const String actionButtonText = "Action";
+
+    testWidgets('show suggested action dialog', (tester) async {
+      await prepare(tester);
+
+      showSuggestedActionDialog(context: context, title: title, message: message,
+        actionButtonText: actionButtonText, onActionButtonPress: () {});
+      await tester.pump();
+
+      checkSuggestedActionDialog(tester, expectedTitle: title, expectedMessage: message,
+        expectedActionButtonText: actionButtonText);
+    }, variant: const TargetPlatformVariant({TargetPlatform.android, TargetPlatform.iOS}));
+
+    testWidgets('user presses action button', (tester) async {
+      await prepare(tester);
+
+      late String actionText;
+      const String expectedActionText = "Action performed!";
```

Comment:
This can be simplified, because the string isn't doing any work here:
```suggestion
      bool wasPressed = false;
```